spec/Modules.tex

```diff
@@ -98,12 +102,12 @@ \section{Files and Implicit Modules}
 Module MX
 Module MY
 \end{chapeloutput}
-Module MX defines top-level module symbols x and printX, while MY
-defines top-level module symbols y and printY.
+Module MX defines module-scope symbols x and printX, while MY
+defines module-scope symbols y and printY.
```

Comment:
General question: should these be "module-scope*d*" instead?

Reply:
I tried to use "module-scope" whenever it acted as an adjective/modifier and "module scope" in other cases (similar to how I'd say, "low-level language" vs. "at a low level.").

My brain continues to think "module-scope x" sounds better than "module-scoped x" for whatever reason.  By analogy, I'd say "global-scope variable" or "local-scope variable" rather than "global-scoped variable" or "local-scoped variable."

@mppf, do you have an opinion here?

Reply:
I don't have a real strong opinion here but it might be better to just not use language like `module-scope x` and instead say `x is a module-scope variable within MX` e.g.

Reply:
Sorry to be unclear:  When I said "module-scope x" I meant for "x" to stand in for the words "symbol" or "statement."  I take it from your answer that you're OK with "module-scope variable" rather than "module-scoped variable" though, which is what I was trying to get at.

Reply:
I feel consistently about "module-scope x" and "global-scope variable" (as in , I would use "module-scoped" and "global-scoped").  Doing a little bit of digging, I think these are called "deverbal adjectives" and the only examples I am seeing are either in the "-ed" or "-ing" form (https://en.wikipedia.org/wiki/Attributive_verb).

Reply:
Googling, it looks like this may just be another example of computer people abusing English:

* "local-scope variable": 9100
* "global-scope variable": 9650
* "global-scoped variable": 669
* "local-scoped variable": 452

But the last time I relied on Google for an argument like this I was accused of not using it right, so you should double-check my work.

Reply:
I'm also realizing that "top-level symbol/statement" (what we had before) is not so grammatically different from "global-scope symbol/statement."  "top", like "local"/"global"/"module" serves as an adjective; "level" like "scope" is a noun.  Together the two form an adjective of sorts to qualify "symbol" or "statement."  So why not "top-leveled symbol/statement"?

Reply:
Hmm, that's a fair point.  But "level" can also be used as an adjective in its own right ("at the same height as someone or something else") in addition to being a verb and a noun, whereas "scope" is listed as either a verb or a noun.

Reply:
(I got that by just googling "scope" and "level").  If that's not convincing, I'll just back down :)

Reply:
I'm not as convinced by English grammar rules when it comes to technical/PL jargon as I am by custom and what seems right, but am willing to admit I could be wrong, which is why I opened the chatroom poll rather than just forging ahead.  It backs the current strategy pretty strongly (6-0 if we discount your vote and my abstention), so given your willingness to back down, I say let's proceed as-is.

Reply:
Sounds good!